### PR TITLE
Refactors sonobuoy into build

### PR DESCRIPTION
**Pull request by chuckha**
Sonobuoy gets refactorted into the build directory.

The top level Makefile now defers build commands to the Makefiles found in the build/ directory.

Signed-off-by: Chuck Ha <chuck@heptio.com>

### CODE REVIEW

**Comment by kensimon** [comment]
run tests

**Comment by ncdc** [comment]
What's the motivation for this?

**Comment by chuckha** [comment]
@ncdc The motivation behind this is to modify the sonobuoy image and have the image run a script exactly like the conformance test and systemd logs plugin do. It would take the results of a sonobuoy run and put a reference to the results directory in the `/done` file. In this way, we could have another container running in the same pod as the master that watches for the `/done` directory and say, uploads results somewhere more permanent.

**Comment by ncdc** [comment]
What about having this in a `build` directory enables that / what about the current structure makes that difficult or impossible?

**Comment by chuckha** [comment]
@ncdc this felt more organized to me as I was planning on adding some more files (like a runner script). It would be nice to keep them all in one directory and not in the top level (or have the build be split across the top level and a subdirectory). Happy to hear alternatives though.

File: .gitignore [hunk 1]
```diff
@@ -1,10 +1,39 @@
...
+*.o
+*.a
+*.so
+
+# Folders
+_obj
+_test
+
+# Architecture specific extensions/prefixes
+*.[568vq]
```
Comment (timothysc) [inline .gitignore]: ?

Comment (chuckha) [inline .gitignore]: Oh, I haven't the foggiest. This is legacy and can instead be merged with what @kensimon added in #1

Comment (kensimon) [inline .gitignore]: Needs a rebase, and I'd say strip this down to what I merged in #1 unless you know you need to gitignore each of these lines.

**Comment by timothysc** [comment]
@ncdc Thoughts from @chuckha 's last comment?

**Comment by ncdc** [comment]
I think that reasoning is reasonable 😄. I was mostly wondering if other projects are going to be required to use this build subdirectory structure, as simpler projects might not need it.

**Comment by timothysc** [comment]
> I was mostly wondering if other projects are going to be required to use this build subdirectory structure, as simpler projects might not need it. 

I think "it depends" on the complexity of the project, and we can rationalize over time.

**Comment by kensimon** [comment]
run tests

**Comment by timothysc** [comment]
@chuckha please update per comments then feel free to self-merge.

**Pull request merged**